web/src/stream_list.js

```diff
@@ -595,14 +595,20 @@ export function get_sidebar_stream_topic_info(filter) {
     return result;
 }
 
-function deselect_stream_items() {
-    $("ul#stream_filters li").removeClass("active-filter");
+function deselect_stream_items(active_stream_id) {
+    $("ul#stream_filters li").removeClass("active-filter stream-expanded");
+    if (active_stream_id) {
+        const active_stream_li = get_stream_li(active_stream_id);
+        if (active_stream_li) {
+            active_stream_li.addClass("stream-expanded");
+        }
+    }
 }
 
 export function update_stream_sidebar_for_narrow(filter) {
     const info = get_sidebar_stream_topic_info(filter);
 
-    deselect_stream_items();
+    deselect_stream_items(info.stream_id);
```

Comment:
Can you explain how you ensure that `stream-expanded` is set whenever the stream is open? I worry that this is going to be incorrect for cases like where the user navigates to a topic via a link, reloading a browser tab already open to the topic ... or where we rerender the sidebar in the background (e.g. because a user subscribed to a new stream).

Also, `deselect_stream_items` seems like a bad choice of function name for where to put this logic -- `update_stream_sidebar_for_narrow` would make more sense, or its own independent function.

Reply:
So `deselect_stream_items` is the only place where we were removing `active-filter` for all stream rows even if it was indeed expanded this includes all the scenarios you mentioned.

But I agree adding this logic here in this function does not make a whole lot of sense and the logic itself is a little bit excessive. I have refactored this change to add `stream-expanded` class in `update_stream_sidebar_for_narrow` function irrespective of whether a topic is narrowed inside that stream row or not.